src/Spice86/MemoryWrappers/InstructionsDecoder.cs

```diff
@@ -81,15 +84,74 @@ public List<CpuInstructionInfo> DecodeInstructions(uint startAddress,
         return instructions;
     }
 
-    private static Decoder InitializeDecoder(CodeReader codeReader, uint currentIp) {
-        Decoder decoder = Decoder.Create(16, codeReader, currentIp,
-            DecoderOptions.Loadall286 | DecoderOptions.Loadall386);
+    /// <summary>
+    /// Enhanced decoder that supports bidirectional decoding and custom formatting for modern views.
+    /// </summary>
+    /// <param name="centerAddress">The address to center the decoding around</param>
+    /// <param name="blockSize">The size of the memory block around the requested address to decode</param>
+    /// <returns>A list of decoded CPU instructions</returns>
+    public Dictionary<uint, EnrichedInstruction> DecodeInstructionsExtended(SegmentedAddress centerAddress, uint blockSize) {
+        uint halfBlockSize = blockSize / 2;
+
+        // Let's always start at the start of the segment
+        uint blockStartAddress = (uint)(centerAddress.Segment << 4);
+        uint length = Math.Min(Math.Max(blockSize, centerAddress.Offset + halfBlockSize), 0xFFFF);
+
+        // Read the memory block
+        byte[] memoryBlock = _memory.ReadRam(length, blockStartAddress);
+
+        // Create a decoder for the memory block
+        var codeReader = new ByteArrayCodeReader(memoryBlock);
+        var decoder = Decoder.Create(16, codeReader);
+
+        // Create a dictionary to hold the instructions
+        var instructions = new Dictionary<uint, EnrichedInstruction>();
+
+        var currentAddress = new SegmentedAddress(centerAddress.Segment, 0);
```

Comment:
Could be created line 97 and then the Linear property could be used instead of computing it twice. Would make intent of blockStartAddress computation clearer.

Reply:
Done